src/screens/scanning/ReviewScreen.tsx

```diff
@@ -116,9 +117,18 @@ export default function ReviewScreen({
       });
     }
 
-    navigation.navigate('ConfirmationScreen', {
-      count: voucherMap.size,
-    });
+    navigation.dispatch(
+      CommonActions.reset({
+        index: 1,
+        routes: [
+          { name: 'VoucherEntryStartScreen' },
+          {
+            name: 'ConfirmationScreen',
+            params: { count: voucherMap.size },
+          },
+        ],
+      }),
+    );
```

Comment:
The edit amount modal is triggering "unhandled promise rejection" warnings, although it doesn't appear to be slowing down the app. Just noticed an odd message in the console.

Reply:
@selene-huang can you look into this?

Reply:
This seems to only be happening when you edit it to an invalid amount and the change gets rejected. We probably aren't catching something, but it doesn't seem to affect functionality too much.

Reply:
Should be handled now.